pkg/server/gitproviders/gitprovider.go

```diff
@@ -22,31 +23,51 @@ func (s *GitProviderService) GetGitProviderForUrl(repoUrl string) (gitprovider.G
 	for _, p := range gitProviders {
 		gitProvider, err := s.GetGitProvider(p.Id)
 		if err != nil {
-			return nil, "", err
+			continue
 		}
 
 		canHandle, _ := gitProvider.CanHandle(repoUrl)
 		if canHandle {
-			return gitProvider, p.Id, nil
+			_, err = gitProvider.GetRepositoryContext(gitprovider.GetRepositoryContext{
+				Url: repoUrl,
+			})
+			if err == nil {
+				return gitProvider, p.Id, nil
+			}
 		}
 	}
-
-	u, err := url.Parse(repoUrl)
-	if err != nil {
-		return nil, "", nil
+	supportedGitProviders := []string{
```

Comment:
You can import this from `github.com/daytonaio/daytona/cmd/daytona/config` as `config.GetSupportedGitProviders()`. Hardcoding this here is a bit fragile.